src/rpc/common/impl/Processors.h

```diff
@@ -31,21 +31,61 @@ struct DefaultProcessor final
     operator()(HandlerType const& handler, boost::json::value const& value)
         const
     {
-        using boost::json::value_from;
-        using boost::json::value_to;
+        // can not do the check in one if statement
+        // first check the nested type exist
+        // then check has a process function called by this type
+        constexpr bool hasNonVoidInputField =
+            requires(const HandlerType& t) { typename HandlerType::Input; };
+        if constexpr (hasNonVoidInputField)
+        {
+            constexpr bool hasProcessWithInput =
+                requires(const HandlerType& t, typename HandlerType::Input in) {
+                    t.process(in);
+                };
+            if constexpr (hasProcessWithInput)
+            {
+                using boost::json::value_from;
+                using boost::json::value_to;
 
-        // first we run validation
-        auto const spec = handler.spec();
-        if (auto const ret = spec.validate(value); not ret)
-            return Error{ret.error()};  // forward Status
+                // first we run validation
+                auto const spec = handler.spec();
+                if (auto const ret = spec.validate(value); not ret)
+                    return Error{ret.error()};  // forward Status
 
-        auto const inData = value_to<typename HandlerType::Input>(value);
+                auto const inData =
+                    value_to<typename HandlerType::Input>(value);
 
-        // real handler is given expected Input, not json
-        if (auto const ret = handler.process(inData); not ret)
-            return Error{ret.error()};  // forward Status
+                // real handler is given expected Input, not json
+                if (auto const ret = handler.process(inData); not ret)
+                    return Error{ret.error()};  // forward Status
+                else
+                    return value_from(ret.value());
+            }
+        }
         else
-            return value_from(ret.value());
+        {
+            return Error{RPC::Status{"Wrong call"}};
```

Comment:
You can use the static_assert trick here to make this fail to compile instead of a runtime error:

```cpp
template <typename>
static constexpr bool unsupported_v = false;

// down the line ...
else { static_assert(unsupported_v<Type>); }
```

Reply:
I think we can't have compile error here. Because I add the process() in Model class and AnyHandler to support no input process. So DefaultProcessor need valid method definition for both process(input) and proces() . If we wanna constrain it in compile time, we have to more changes on AnyHandler / Model